src/pages/view-package.tsx

```diff
@@ -1,24 +1,57 @@
 import RepoPageContent from "@/components/ViewPackagePage/components/repo-page-content"
 import { useCurrentPackageInfo } from "@/hooks/use-current-package-info"
-import { usePackageByName } from "@/hooks/use-package-by-package-name"
 import { usePackageFiles } from "@/hooks/use-package-files"
 import { usePackageRelease } from "@/hooks/use-package-release"
 import { useLocation, useParams } from "wouter"
 import { Helmet } from "react-helmet-async"
+import { useEffect, useState } from "react"
+import NotFoundPage from "./404"
 
 export const ViewPackagePage = () => {
-  const { packageInfo } = useCurrentPackageInfo()
+  const {
+    packageInfo,
+    isLoading: isLoadingPackageInfo,
+    error: packageInfoError,
+  } = useCurrentPackageInfo()
   const { author, packageName } = useParams()
   const [, setLocation] = useLocation()
+  const [isNotFound, setIsNotFound] = useState(false)
 
-  const { data: packageRelease } = usePackageRelease({
+  const {
+    data: packageRelease,
+    isLoading: isLoadingPackageRelease,
+    error: packageReleaseError,
+  } = usePackageRelease({
     is_latest: true,
     package_name: `${author}/${packageName}`,
   })
 
   const { data: packageFiles } = usePackageFiles(
     packageRelease?.package_release_id,
   )
+  useEffect(() => {
+    if (!isLoadingPackageInfo && !isLoadingPackageRelease) {
+      if (
+        packageInfoError?.status === 404 ||
+        packageReleaseError?.status === 404 ||
+        (!packageInfo && !isLoadingPackageInfo) ||
+        (!packageRelease && !isLoadingPackageRelease)
+      ) {
+        setIsNotFound(true)
```

Comment:
Use early return instead of neated if conditions, you can also use a lot of if conditions, this makes it easier to read

Reply:
*nested